gitlab/misc.sh

```diff
@@ -14,24 +14,14 @@ function load_ssh_agent() {
 
 function validate_schemas_kubecfg() {
   local dir
-  dir=${1-"environments"}
+  dir=${1-"rendered"}
 
-  for cluster in $(find $dir -type f -name kustomization.yaml -exec dirname {} \;)
+  for cluster in $(find $dir -type d -links 2)
```

Comment:
what does this do ? does not work on my laptop :)

would not be better to just search for a `well known file` like `ClusterIssuer-selfsigning-issuer.yaml` ?

Reply:
It finds all directories that don't contain any other directories (the 2 links are for `.` and `..` respectively, which every directory has). Works on my machine, plus on Alpine for the GitLab runners!

Looking for a specific file would be problematic given that we're running this in two different repos (for Kustomize and Kubecfg)

Reply:
mmmhhh ... i am not really sure about this.

first of all `.` and `..` are not links but directories , https://unix.stackexchange.com/questions/63081/what-are-and-directories 

```
➜  /tmp  file .  
.: sticky, directory
➜  /tmp  file .. 
..: directory
➜  /tmp  file testlink
testlink: symbolic link to terraform-log309896659
```

Also seems very weak.

I say, either keep 2 separate loops , in one check for the `ClusterIssuer-selfsigning-issuer.yaml` and the other for `config.yaml` or revert this bit and let's re-render so we can go on with the real task and we'll revisit later ... i doubt it adds that much latency to the whole job anyway compared to the crappiness of our gitlab runners anyway

Reply:
They're hard links not symlinks. The first and second answers in that post agree with me :)

First answer:
> Notice that bin and sbin each has two links to it. One is the directory entry in the root directory, and the other is the . entry inside that directory.

Second answer:
> When a directory B is created in Unix, it is added as a new entry to another directory A (its parent directory), and in B, two entries are added: one called . as a hard link to itself, and one called .. as a hard link to A.

It was just a quick way of finding bottom-level directories which should in theory contain rendered manifests. I can split it back out into 2 if you're really convinced that's better, it just feels less flexible to me, and it means we need to change it if we change file names later on.

Reply:
I guess we don't know for certain that every bottom-level directory would definitely contain what we expect, but I don't see it as very likely that this way would break.

Reply:
Well it is still not working on my workstation so there is definetely more to this.

LEt's just revert to rendering and go on with the real task ?

Reply:
Have changed it to find the well-known files like you suggested